src/pkg/boltz/boltzService.ts

```diff
@@ -0,0 +1,372 @@
+    network: NetworkString,
+    invoiceAmount: number
+  ): Promise<ReverseSwap>;
+  makeClaimTransaction(p: MakeClaimTransactionParams): Promise<Transaction>;
+  getInvoiceExpireDate(invoice: string): number;
+  getInvoiceValue(invoice: string): number;
+}
+
+export interface SubmarineSwap {
+  address: string;
+  expectedAmount: number;
+  id: string;
+  redeemScript: string;
+}
+
+export interface ReverseSwap {
+  blindingPrivateKey: string;
+  claimPublicKey: string;
+  id: string;
+  invoice: string;
+  lockupAddress: string;
+  preimage: Buffer;
+  redeemScript: string;
+  timeoutBlockHeight: number;
+}
+
+interface MakeClaimTransactionParams {
+  utxo: Unspent;
+  claimPublicKey: Buffer;
+  claimKeyPair: ECPairInterface;
+  preimage: Buffer;
+  redeemScript: Buffer;
+  destinationScript: Buffer;
+  fee: number;
+  password: string;
+  blindingPublicKey: Buffer;
+  timeoutBlockHeight: number;
+}
+
+const feeAmount = 500; // fee for regular liquid tx
+const swapFeeAmount = 500; // fee for Boltz
+// lightning swap invoice amount limit (in satoshis)
+export const DEFAULT_LIGHTNING_LIMITS = { maximal: 4294967, minimal: 50000 };
+export const DEPOSIT_LIGHTNING_LIMITS = {
+  maximal: DEFAULT_LIGHTNING_LIMITS.maximal - feeAmount - swapFeeAmount,
+  minimal: DEFAULT_LIGHTNING_LIMITS.minimal - feeAmount - swapFeeAmount,
+};
+
+export class BoltzService implements BoltzServiceInterface {
+  private boltz: Boltz;
+  private network: NetworkString;
+
+  constructor(network: NetworkString) {
+    this.network = network;
+    this.boltz = new Boltz(network);
+  }
+
+  async getBoltzPair(pair: string): Promise<any> {
+    return this.boltz.getPair(pair);
+  }
+
+  // return invoice expire date
+  getInvoiceExpireDate(invoice: string): number {
+    const { timeExpireDate } = bolt11.decode(invoice);
+    return timeExpireDate ? timeExpireDate * 1000 : 0; // milliseconds
+  }
+
+  // return value in given invoice
+  getInvoiceValue(invoice: string): number {
+    const { satoshis, millisatoshis } = bolt11.decode(invoice);
+    if (satoshis) return fromSatoshi(satoshis, 8);
+    if (millisatoshis) return fromSatoshi(Number(millisatoshis) / 1000, 8);
+    return 0;
+  }
+
+  async makeClaimTransaction({
+    utxo,
+    claimPublicKey,
+    claimKeyPair,
+    preimage,
+    redeemScript,
+    destinationScript,
+    fee,
+    password,
+    blindingPublicKey,
+    timeoutBlockHeight,
+  }: MakeClaimTransactionParams): Promise<Transaction> {
+    const pset = Creator.newPset();
+    const updater = new Updater(pset);
+
+    pset.addInput(
+      new CreatorInput(utxo.txid, utxo.vout, 0xfffffffd, timeoutBlockHeight).toPartialInput()
+    );
+    updater.addInSighashType(0, Transaction.SIGHASH_ALL);
+    updater.addInWitnessUtxo(0, utxo.witnessUtxo!);
+    updater.addInWitnessScript(0, utxo.witnessUtxo!.script);
+
+    updater.addOutputs([
+      {
+        script: destinationScript,
+        blindingPublicKey,
+        asset: networks[this.network].assetHash,
```

Comment:
let's pass an `asset` argument and remove network member ? (delete a deps on networks)

Reply:
Boltz class still need the network for
```
export const boltzUrl: Record<NetworkString, string> = {
  regtest: 'http://localhost:9090',
  testnet: 'https://testnet.boltz.exchange/api',
  liquid: 'https://boltz.exchange/api',
};
```

Unless it takes `boltzUrl` instead